From da1da18e246c6a1951e6f65ba2c4083e86296069 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Fri, 4 Mar 2022 19:26:53 -0500
Subject: [PATCH 30/35] Fix FOD on ZF8

Goodix HAL expects events in this order:
- Fingerprint down when touchscreen says fp down
- UI Ready when HWC says HBM on
- Fingerprint up

Not UI Ready then fingerprint down like what was previously implemented
---
 .../android/server/biometrics/AuthService.java   | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/services/core/java/com/android/server/biometrics/AuthService.java b/services/core/java/com/android/server/biometrics/AuthService.java
index 718dbf85bdcd..f49b73f6922a 100644
--- a/services/core/java/com/android/server/biometrics/AuthService.java
+++ b/services/core/java/com/android/server/biometrics/AuthService.java
@@ -107,6 +107,7 @@ public class AuthService extends SystemService {
     final IAuthService.Stub mImpl;
 
     private FileObserver fodFileObserver = null;
+    private FileObserver fodFileObserver2 = null;
     private ISehBiometricsFingerprint mSamsungFingerprint = null;
 
     /**
@@ -693,19 +694,22 @@ public class AuthService extends SystemService {
             };
             fodFileObserver.startWatching();
         }
-        String asusSpotOnAchieved = "/sys/class/drm/spot_on_achieved";
-        if( (new File(asusSpotOnAchieved)).exists()) {
-            fodFileObserver = new FileObserver(asusSpotOnAchieved, FileObserver.MODIFY) {
+
+        String asusGhbmOnAchieved = "/sys/class/drm/ghbm_on_achieved";
+        if( (new File(asusGhbmOnAchieved)).exists()) {
+            fodFileObserver2 = new FileObserver(asusGhbmOnAchieved, FileObserver.MODIFY) {
                 boolean wasOn = false;
                 @Override
                 public void onEvent(int event, String path) {
-                    String spotOn = readFile(asusSpotOnAchieved);
+                    String spotOn = readFile(asusGhbmOnAchieved);
                     if("1".equals(spotOn)) {
                         if(!wasOn) {
                             try {
                                 IGoodixFingerprintDaemon goodixDaemon = IGoodixFingerprintDaemon.getService();
+
+                                //Send UI ready
                                 goodixDaemon.sendCommand(200002, new java.util.ArrayList<Byte>(), (returnCode, resultData) -> {
-                                    Slog.e(TAG, "Goodix send command returned code "+ returnCode);
+                                    Slog.e(TAG, "Goodix send command touch pressed returned code "+ returnCode);
                                 });
                             } catch(Throwable t) {
                                 Slog.d("PHH-Enroll", "Failed sending goodix command", t);
@@ -717,7 +721,7 @@ public class AuthService extends SystemService {
                     }
                 }
             };
-            fodFileObserver.startWatching();
+            fodFileObserver2.startWatching();
         }
     }
 
-- 
2.25.1

